### DISCUSSION TITLE
Minio client copy with continue seems not working

### DISCUSSION THREAD

**Discussion by ghost** [minio/minio#13471] [Q&A]
Hello,

I'm currently assessing Minio Server / Minio Client for a project and the cp resume management is a very important feature.

I did several tests but it seems that it is not working (when I relaunch the copy, it restart from the start and the incomplete file does not appear with ls -incomplete command). I checked in the documentation but I was not able to find any specific configuration on server side to enable the feature.

The command I use to copy is the following : "mc.exe cp --insecure --continue D:/minio/2.iso mcgAccess/testbucket"

I'm using minio server version 2021-10-13T00:23:17Z and minio client version  2021-10-07T04-19-58Z.

Can someone explain me if I'm missing something?

Thanks in advance for your help.

**Comment by harshavardhana** [comment]
> Hello,
> 
> I'm currently assessing Minio Server / Minio Client for a project and the cp resume management is a very important feature.
> 
> I did several tests but it seems that it is not working (when I relaunch the copy, it restart from the start and the incomplete file does not appear with ls -incomplete command). I checked in the documentation but I was not able to find any specific configuration on server side to enable the feature.
> 
> The command I use to copy is the following : "mc.exe cp --insecure --continue D:/minio/2.iso mcgAccess/testbucket"
> 
> I'm using minio server version 2021-10-13T00:23:17Z and minio client version  2021-10-07T04-19-58Z.
> 
> Can someone explain me if I'm missing something?
> 
> Thanks in advance for your help.

--continue is for resuming when you are uploading multiple files.

It will not resume single uploads.

**Reply by ghost** [reply]
Thanks for your answer! So this means that there is no way to resume a big single upload?

**Reply by harshavardhana** [reply]
> Thanks for your answer! So this means that there is no way to resume a big single upload?

That does not work safely because of AWS S3 behavior https://github.com/aws/aws-sdk-go/issues/1518

**Reply by aisensiy** [reply]
So what if I have a super big file and it failed to upload the full parts?

**Reply by klauspost** [reply]
@aisensiy Retry the upload. It is not a feature we are looking to implement, if that is what you are asking.

You can also see if other S3 clients support resuming individual files. Unfortunately I do not know anyone off hand.

**Reply by aisensiy** [reply]
@klauspost Thanks for your quick reply. But retry the upload means to upload it from zero point? This sounds not very user friendly...

**Reply by harshavardhana** [reply]
@aisensiy feel free to write your own, read the APIs.
